src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -42,23 +44,92 @@ bool Series::validate() const
+    {
+        logs.warning() << "Values for PMAX series should be positive";
+        return false;
+    }
+
+    auto checkVectBetween = [](std::vector<double> v) {
+        return std::count_if(v.begin(), v.end(), [](double d){ return (d >= 0.0 && d <= 1.0); });
+    };
+    if (!checkVectBetween(lowerRuleCurve) || !checkVectBetween(upperRuleCurve))
+    {
+        logs.warning() << "Values for rule curve series should be between 0 and 1";
+        return false;
+    }
+
     return true;
 }
 
+/* bool Series::loadFromFolder(const std::string& folder) */
+/* { */
+/*     bool ret = true; */
+/* #define SEP Yuni::IO::Separator */
+/*     ret = loadVector(folder + SEP + "PMAX-injection.txt", maxInjectionModulation) && ret; */
+/*     ret = loadVector(folder + SEP + "PMAX-withdrawal.txt", maxWithdrawalModulation) && ret; */
+/*     ret = loadVector(folder + SEP + "inflows.txt", inflows) && ret; */
+/*     ret = loadVector(folder + SEP + "lower-rule-curve.txt", lowerRuleCurve) && ret; */
+/*     ret = loadVector(folder + SEP + "upper-rule-curve.txt", upperRuleCurve) && ret; */
+/* #undef SEP */
+/*     return ret; */
+/* } */
+
 bool Series::loadFromFolder(const std::string& folder)
 {
     bool ret = true;
 #define SEP Yuni::IO::Separator
-    ret = loadVector(folder + SEP + "PMAX-injection.txt", maxInjectionModulation) && ret;
-    ret = loadVector(folder + SEP + "PMAX-withdrawal.txt", maxWithdrawalModulation) && ret;
-    ret = loadVector(folder + SEP + "inflows.txt", inflows) && ret;
-    ret = loadVector(folder + SEP + "lower-rule-curve.txt", lowerRuleCurve) && ret;
-    ret = loadVector(folder + SEP + "upper-rule-curve.txt", upperRuleCurve) && ret;
+    ret = loadFile(folder + SEP + "PMAX-injection.txt", maxInjectionModulation) && ret;
+    ret = loadFile(folder + SEP + "PMAX-withdrawal.txt", maxWithdrawalModulation) && ret;
+    ret = loadFile(folder + SEP + "inflows.txt", inflows) && ret;
+    ret = loadFile(folder + SEP + "lower-rule-curve.txt", lowerRuleCurve) && ret;
+    ret = loadFile(folder + SEP + "upper-rule-curve.txt", upperRuleCurve) && ret;
 #undef SEP
     return ret;
 }
 
-bool Series::loadVector(const std::string& path, std::vector<double>& vect) const
+bool loadFile(const std::string& path, std::vector<double>& vect)
+{
+    logs.debug() << "Loading file " << path;
+
+    vect.reserve(HOURS_PER_YEAR);
+
+    std::ifstream file;
+    file.open(path);
+
+    if (!file)
+    {
+        logs.debug() << "File not found: " << path;
+        return true;
+    }
+
+    std::string line;
+    try
+    {
+        unsigned int count = 0;
+        while (getline(file, line) && count < HOURS_PER_YEAR)
+        {
+            double d = std::stod(line);
+            vect.push_back(d);
+            count++;
+        }
+        if (count < HOURS_PER_YEAR)
+            return false;
+    }
+    catch (const std::exception&)
+    {
+        logs.error() << "Failed reading file: " << path;
+        return false;
+    }
```

Comment:
Catching a generic exception won't be of much help to the user
```suggestion
    catch (const std::ios_base::failure& ex)
    {
        logs.error() << "Failed reading file: " << path << " (I/O error)";
        return false;
    }
    catch (const std::invalid_argument& ex)
    {
        logs.error() << "Failed reading file: " << path << " conversion to double failed at line " << line;
    }
    catch (const std::out_of_range& ex)
    {
        logs.error() << "Failed reading file: " << path << " value is out of bounds at line " << line;
    } 
```